apprise/plugins/gatewayapi.py

```diff
@@ -0,0 +1,312 @@
+        **{
+            "apikey": {
+                "name": _("API Key"),
+                "type": "string",
+                "private": True,
+                "required": True,
+            },
+            "target_phone": {
+                "name": _("Target Phone No"),
+                "type": "string",
+                "prefix": "+",
+                "regex": (r"^[0-9\s)(+-]+$", "i"),
+                "map_to": "targets",
+            },
+            "targets": {
+                "name": _("Targets"),
+                "type": "list:string",
+                "required": True,
+            },
+        },
+    )
+
+    # Define our template arguments
+    template_args = dict(
+        NotifyBase.template_args,
+        **{
+            "key": {
+                "alias_of": "apikey",
+            },
+            "to": {
+                "alias_of": "targets",
+            },
+            "from": {
+                "name": _("From Phone No/Sender"),
+                "type": "string",
+            },
+        },
+    )
+
+    def __init__(self, apikey=None, targets=None, source=None, **kwargs):
+        """Initialize GatewayAPI Object."""
+        super().__init__(**kwargs)
+
+        self.apikey = validate_regex(apikey)
+        if not self.apikey:
+            msg = f"An invalid API Key ({apikey}) was specified."
+            self.logger.warning(msg)
+            raise TypeError(msg)
+
+        # The sender (optional)
+        self.source = source
+
+        # Parse our targets
+        self.targets = []
+
+        has_error = False
+        for target in parse_phone_no(targets):
+            # Parse each phone number we found
+            result = is_phone_no(target)
+            if result:
+                self.targets.append(result["full"])
+                continue
+
+            has_error = True
+            self.logger.warning(
+                f"Dropped invalid phone # ({target}) specified.",
+            )
+
+        if not self.targets and has_error:
+            msg = "No valid phone numbers were specified."
+            self.logger.warning(msg)
+            raise TypeError(msg)
+
+        return
+
+    def send(self, body, title="", notify_type=NotifyType.INFO, **kwargs):
+        """Perform GatewayAPI Notification."""
+
+        if not self.targets:
+            # We have nothing to notify
+            self.logger.warning("There are no GatewayAPI targets to notify")
+            return False
+
+        # error tracking (used for function return)
+        has_error = False
+
+        # Prepare our headers
+        headers = {
+            "User-Agent": self.app_id,
+        }
+
+        # Prepare our targets
+        targets = list(self.targets)
+        while len(targets):
+            # Get our target to notify
+            target = targets.pop(0)
+
+            # Prepare our payload
+            payload = {
+                "message": body,
+                "recipients.0.msisdn": int(target),
+            }
```

Comment:
This is a great start! but when i visit https://gatewayapi.com/docs/apis/rest/#sending-sms-messages, it seems like the payload doesn't align. You're missing quit a few fields.  Perhaps you're a developer of the system and know of some un-documented approaches?

Reply:
Fair enough; get it working for you and I can help you with test cases.